crates/interpreter/src/side_table.rs

```diff
@@ -20,43 +20,37 @@ use crate::module::Parser;
 
 pub struct SideTableView<'m> {
     pub func_idx: usize,
-    pub indices: &'m [u16], // including 0 and the length of metadata_array
-    pub metadata: &'m [u16],
+    pub indices: &'m [u8], // including 0 and the length of metadata_array
+    pub metadata: &'m [u8],
 }
 
 impl<'m> SideTableView<'m> {
-    pub fn new(parser: &mut crate::valid::Parser<'m>) -> Result<Self, Error> {
+    pub fn new(binary: &'m [u8]) -> Result<Self, Error> {
+        let num_functions = parse_u16(binary, 0) as usize;
+        let indices_end = 2 + (num_functions + 1) * 2;
         Ok(SideTableView {
             func_idx: 0,
-            indices: parse_side_table_field(parser)?,
-            metadata: parse_side_table_field(parser)?,
+            indices: &binary[2 .. indices_end],
```

Comment:
The TODO below also applies to this function. In particular, we should use `.get(2 .. indices_end)` to return an error instead of panicking.